### DISCUSSION TITLE
postcss-px-to-viewport 不能对vant动态计算的尺寸单位进行转换吗？

### DISCUSSION THREAD

**Discussion by jianfei0220** [youzan/vant#12401] [Q&A]
请问有解决方案吗？

**Comment by chenjiahan** [comment]
postcss-px-to-viewport 只能转换 CSS 代码里的单位，没法处理通过 JS 设置的单位。

Vant 里通过 JS 设置的单位一般都有提供 props，大部分都可以直接传入 vh/vw 单位，你可以通过 props 来控制。

**Comment by jianfei0220** [comment]
&nbsp;比如这个picker，自带的样式并没有进行转换。




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "youzan/vant"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年10月29日(星期天) 下午5:54
***@***.***&gt;;
抄送:&nbsp;"_ ***@***.******@***.***&gt;;
主题:&nbsp;Re: [youzan/vant] postcss-px-to-viewport 不能对vant动态计算的尺寸单位进行转换吗？ (Discussion #12401)





 
postcss-px-to-viewport 只能转换 CSS 代码里的单位，没法处理通过 JS 设置的单位。
 
Vant 里通过 JS 设置的单位一般都有提供 props，大部分都可以直接传入 vh/vw 单位，你可以通过 props 来控制。

**Reply by inottn** [reply]
picker 的 option-height 支持 px vw vh rem 单位，你可以自定义试试

**Comment by jianfei0220** [comment]
好的 感谢&nbsp;





&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年10月30日(星期一) 中午11:04
收件人: ***@***.***&gt;; 
抄送: "_ ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [youzan/vant] postcss-px-to-viewport 不能对vant动态计算的尺寸单位进行转换吗？ (Discussion #12401)





 
picker 的 option-height 支持 px vw vh rem 单位，你可以自定义试试
